src/query/expression/src/types/bitmap.rs

```diff
@@ -36,18 +38,214 @@ use crate::values::Scalar;
+        for value in &data {
+            value.serialize_into(&mut raw.data).unwrap();
+            raw.commit_row();
+        }
+        Self {
+            raw: raw.build(),
+            data,
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.data.len()
+    }
+
+    pub fn index(&self, index: usize) -> Option<&HybridBitmap> {
+        self.data.get(index)
+    }
+
+    #[allow(clippy::missing_safety_doc)]
+    pub unsafe fn index_unchecked(&self, index: usize) -> &HybridBitmap {
+        self.data.get_unchecked(index)
+    }
+
+    pub fn slice(&self, range: Range<usize>) -> Self {
+        let raw = self.raw.slice(range.clone());
+        let data = self.data[range].to_vec();
+        Self { raw, data }
+    }
+
+    pub fn iter(&self) -> BitmapColumnIter<'_> {
+        BitmapColumnIter {
+            inner: self.data.iter(),
+        }
+    }
+
+    pub fn raw(&self) -> &BinaryColumn {
+        &self.raw
+    }
+
+    pub fn into_raw(self) -> BinaryColumn {
+        self.raw
+    }
+
+    pub fn memory_size(&self) -> usize {
+        let raw = self.raw.data().len() + self.raw.offsets().len() * 8;
+        let decoded = self.data.iter().map(|b| b.memory_size()).sum::<usize>();
+        raw + decoded
+    }
+
+    pub fn check_valid(&self) -> Result<()> {
+        Ok(self.raw.check_valid()?)
+    }
+
+    pub fn data(&self) -> &[u8] {
+        self.raw.data()
+    }
+
+    pub fn offsets(&self) -> &[u64] {
+        self.raw.offsets()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct BitmapColumnIter<'a> {
+    inner: std::slice::Iter<'a, HybridBitmap>,
+}
+
+impl<'a> Iterator for BitmapColumnIter<'a> {
+    type Item = &'a HybridBitmap;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.next()
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.inner.size_hint()
+    }
+}
+
+unsafe impl<'a> TrustedLen for BitmapColumnIter<'a> {}
+
+#[derive(Debug, Clone)]
+pub struct BitmapColumnBuilder {
+    raw: BinaryColumnBuilder,
+    data: Vec<HybridBitmap>,
+}
+
+impl BitmapColumnBuilder {
+    pub fn with_capacity(capacity: usize, raw_capacity: usize) -> Self {
+        Self {
+            raw: BinaryColumnBuilder::with_capacity(capacity, raw_capacity),
+            data: Vec::with_capacity(capacity),
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.data.len()
+    }
+
+    pub fn memory_size(&self) -> usize {
+        let raw = self.raw.memory_size();
+        let decoded = self.data.iter().map(|b| b.memory_size()).sum::<usize>();
+        raw + decoded
+    }
+
+    pub fn push(&mut self, item: &HybridBitmap) {
+        item.serialize_into(&mut self.raw.data).unwrap();
+        self.raw.commit_row();
+        self.data.push(item.clone());
+    }
+
+    pub fn push_repeat(&mut self, item: &HybridBitmap, n: usize) {
+        let mut buf = Vec::new();
+        item.serialize_into(&mut buf).unwrap();
+        self.raw.push_repeat(&buf, n);
```

Comment:
Maybe we should not include `raw` field in builder and column.

It will lead to extra memory allocation and usage.